app/src/main/java/org/dharmaseed/android/PlaybackService.java

```diff
@@ -0,0 +1,232 @@
+import androidx.media3.datasource.DefaultDataSource;
+import androidx.media3.datasource.DefaultHttpDataSource;
+import androidx.media3.exoplayer.ExoPlayer;
+import androidx.media3.exoplayer.source.DefaultMediaSourceFactory;
+import androidx.media3.session.CommandButton;
+import androidx.media3.session.DefaultMediaNotificationProvider;
+import androidx.media3.session.MediaNotification;
+import androidx.media3.session.MediaSession;
+import androidx.media3.session.MediaSessionService;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.text.SimpleDateFormat;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.Timer;
+import java.util.Vector;
+
+public class PlaybackService extends MediaSessionService {
+
+    private static final String LOG_TAG = "PlaybackService";
+    private MediaSession mediaSession;
+    private Runnable updateTalkProgressTask;
+    private Handler handler;
+
+    @Override
+    @OptIn(markerClass = UnstableApi.class)
+    public void onCreate() {
+        super.onCreate();
+
+        // Create the media player
+        Context context = this;
+        DefaultHttpDataSource.Factory httpFactory = new DefaultHttpDataSource.Factory()
+                .setAllowCrossProtocolRedirects(true);
+        DefaultDataSource.Factory dataSourceFactory = new DefaultDataSource.Factory(
+                context, httpFactory
+        );
+
+        AudioAttributes audioAttributes = new AudioAttributes.Builder()
+                .setUsage(C.USAGE_MEDIA)
+                .build();
+
+        ExoPlayer mediaPlayer = new ExoPlayer.Builder(context)
+                .setMediaSourceFactory(
+                        new DefaultMediaSourceFactory(context)
+                                .setDataSourceFactory(dataSourceFactory))
+                .setWakeMode(WAKE_MODE_NETWORK)
+                .setHandleAudioBecomingNoisy(true)
+                .setUseLazyPreparation(true)
+                .setAudioAttributes(audioAttributes, true)
+                .build();
+
+        // Create a media session
+        mediaSession = new MediaSession.Builder(this, mediaPlayer)
+                .setCallback(new SessionCallback())
+                .build();
+
+        // Set a custom media notification
+        setMediaNotificationProvider(new NotificationProvider(this));
+
+        // Set a timer to periodically update the talk progress in the database
+        handler = new Handler();
+        updateTalkProgressTask = new Runnable() {
+            @Override
+            public void run() {
+                updateTalkProgress();
+            }
+        };
+        handler.postDelayed(updateTalkProgressTask, 10000);
+    }
+
+    // Periodic task that saves talk progress in the database
+    private void updateTalkProgress() {
```

Comment:
We need to update the talk progress from the inside this new service instead of in `PlayTalkActivity` as before because playback can continue in the background even when no `PlayTalkActivities` are active.